### ISSUE TITLE
Build/run option to install the community edition (CE) without the EE code

### ISSUE THREAD

**Issue by coglmn** [activepieces/activepieces#3473]
**Is your feature request related to a problem? Please describe.**

Yes, would like to install the standard community edition version of activePieces in a self-hosted mode without the EE code base (the parts of the code in the rep that is subject to enterprise license) to see what is possible to do without the enterprise-graded code. 

Currently, if all files/folders what is under 'ee' identified folders are removed, the compile / build fails with the following error:

 The following implicitDependencies point to non-existent project(s):
      ui-feature-builder-header
       ee-components
 

**Describe the solution you'd like**

As an open-source project, the expectation is to have a build option that allows installing only the community edition code, and EE codes to be add-ons.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

Thank you for the great project!

**Comment by abuaboud** [comment]
Hi @coglmn,

I know that currently, there are still references to EE in the community edition. However, these references are protected by the environment variable AP_EDITION along with the license key. So, if you use it in the usual way, there won't be any code called from EE in the community edition which still a community edition since no code are being used under ee.

Ideally, it would be better to follow your description, but we haven't yet figured out how to 100% inject the ee code code into community code without being imported from community edition, we got closer but not there yet.

Thank you,
Mo.

**Comment by coglmn** [comment]
Hi @abuaboud 

Thank you for the comment, appreciate it.  So, it means that while the code in EE folders exists, they are not called, and the installation would run the CE code only? Therefore, the code is considered CE without having a license in place?

There are notions of conditional and dynamic imports (checks if certain conditions are met and libraries exist to import), not sure if they can help here, though.

Thank you.

**Comment by abuaboud** [comment]
Exactly! feel free to install and use activepieces by default it's CE and licensed under MIT!

You can only use EE by providing valid license key and changing edition which has to be explicit process so you don't have to worry about using EE.

**Comment by madjam002** [comment]
You could do what Posthog do and have an activepieces repo where all the development and discussion takes place, and then a activepieces-foss repo which is a read only mirror with all of the proprietary code automatically removed.

https://github.com/PostHog/posthog
https://github.com/PostHog/posthog-foss

**Comment by abuaboud** [comment]
Hi @madjam002,

great suggestion II think we should do that but currently, it's not at that stage yet. We have removed a couple more references, but there are 4 major files that need to be refactored in the backend so we can inject the references instead of the other way around

**Comment by jficz** [comment]
> Exactly! feel free to install and use activepieces by default it's CE and licensed under MIT!
> 
> You can only use EE by providing valid license key and changing edition which has to be explicit process so you don't have to worry about using EE.

Maybe put this somewhere in the docs explicitly? I just spent some time looking through the `ee` directories trying to get an idea what I need to be careful about. Until I read this disclaimer, I practically considered it a legal minefield and was just about to cross this project out of my evaluation list altogether because of it.

**Comment by linh-pham-codeleap** [comment]
Hi @abuaboud ,

I took a look at the source code and noticed that `packages/server/api/src/app/ee/` is indeed restricted based on `AP_EDITION`.
However, the `packages/ee` directory is used quite extensively in `packages/react-ui` without any restriction.

So, I’m wondering, is it safe to use `packages/react-ui` under the MIT license even though it includes EE components?

<img width="534" height="847" alt="Image" src="https://github.com/user-attachments/assets/2d2e38c1-f98e-4e2c-b49a-facc37feac4c" />

<img width="538" height="504" alt="Image" src="https://github.com/user-attachments/assets/144e5828-21aa-42ca-9147-64245770af02" />

**Comment by ofirattia** [comment]
Do you have the list of the routes that need to be removed on the UI that are related to EE?

**Comment by linh-pham-codeleap** [comment]
Hi @ofirattia ,
I don’t have a complete list, but here’s one example:
The Alert component calls the `v1/alerts` endpoint, which is located under the EE folder.
However, when I run the project under the Community Edition, I can still access the UI path `projects/:project_id/settings/alerts` and perform actions such as listing and creating alerts.

Doesn’t this violate the EE license?

**Comment by ofirattia** [comment]
I decided to leave it, its seems like the components are mixed between ee to the MIT intentionally.

**Comment by maor-rozenfeld** [comment]
Unfortunately this mess prevents companies from contributing back to ActivePieces from their forks. If you want to build on top of ActivePieces, you essentially have to fork it once, fix the ee mess, and never look back. Very unfortunate, but we had no choice.

@ofirattia Feel free to fork our [Apache based repo](https://github.com/openops-cloud/openops) if it's any help.

**Comment by ofirattia** [comment]
> Unfortunately this mess prevents companies from contributing back to ActivePieces from their forks. If you want to build on top of ActivePieces, you essentially have to fork it once, fix the ee mess, and never look back. Very unfortunate, but we had no choice.
> 
> [@ofirattia](https://github.com/ofirattia) Feel free to fork our [Apache based repo](https://github.com/openops-cloud/openops) if it's any help.

Sent you msg on linkedin
